src/core/etl/src/Flow/ETL/PHP/Type/Native/ScalarType.php

```diff
@@ -96,11 +96,6 @@ public function isValid(mixed $value) : bool
         };
     }
 
-    public function isValidArrayKey() : bool
```

Comment:
This check has no point as in PHP:
> The (array) key can either be an int or a string.

Reply:
yeah, but its too late at that point as php will throw exception, and this lets us know upfront

Reply:
At this point we are already with a PHP array, thus we will have int or string, and this check doesn't really do anything, those `ArrayContentDetect` works on the PHP array, so it cannot have type of keys other than int or string.